From: Samrat Kannikar <ksamrat@novell.com>
Subject: novfs: Remove dcache count restricting code
References: bnc#576026
Patch-mainline: no

Attached patch removes the code which restricts the
number of dir_cache entries, that are maintained by novfs

Signed-off-by: Samrat Kannikar <ksamrat@novell.com>
Acked-by: Jan Kara <jack@suse.cz>

diff --git linux-2.6.32-sle11-sp1/fs/novfs/inode.c linux-2.6.32-sle11-sp1/fs/novfs/inode.c
--- linux-2.6.32-sle11-sp1/fs/novfs/inode.c
+++ linux-2.6.32-sle11-sp1/fs/novfs/inode.c
@@ -4346,8 +4346,6 @@ int novfs_add_inode_entry(struct inode *i,
 	struct inode_data *id;
 	struct novfs_dir_cache *new;
 	int retVal = -ENOMEM;
-	struct novfs_dir_cache *todel;
-	struct list_head *todeltmp;
 
 	//SClark
 	DbgPrint("i: %p", i);
@@ -4383,19 +4381,6 @@ int novfs_add_inode_entry(struct inode *i,
 			memcpy(new->name, name->name, name->len);
 			new->name[new->nameLen] = '\0';
 			list_add(&new->list, &id->DirCache);
-
-			if (id->cntDC > 20) {
-				todeltmp = id->DirCache.prev;
-				todel = list_entry(todeltmp, struct novfs_dir_cache, list);
-
-				list_del(&todel->list);
-
-				kfree(todel);
-
-				DCCount--;
-				id->cntDC--;
-			}
-
 		}
 	}
 	return (retVal);
-- 
1.6.4.2
